### ISSUE TITLE
Vaccination Certificate in Wallet (iOS)

### ISSUE THREAD

**Issue by kevin-kraus** [corona-warn-app/cwa-wishlist#503]
## Feature description

Show a pass in Wallet on iOS for quick access in certain situations by just double clicking the side- / home button. 

For example if in the future the case arises that you need a proof of vaccination for certain places to enter (e.g. restaurants), you could access the vaccination certificate quicker by using the wallet shortcuts than searching the CWA and opening the tab for the certificate there. 

## Problem and motivation

User acceptance, simplicity of use and speed of processing for entry controls increase.

## Is this something you're interested in working on

No

**Comment by Ein-Tim** [comment]
I'd suggest to open this issue in https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios

**Comment by jucktnich** [comment]
I do think, that the scanning implementation should be done by the dgca (if it's even possible), but the CWA could add a wallet card by its own.

**Comment by Ein-Tim** [comment]
This is now also tracked in https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69.

**Comment by muuuh** [comment]
Please note that having to unlock the phone for opening the app and potentially handing out the phone is a privacy issue.

**Comment by dsarkar** [comment]
FYI https://github.com/Digitaler-Impfnachweis/certification-apis/discussions/37

**Comment by ezadoo** [comment]
As title of this feature-request says that vaccination certificates should be possible to import into Apple Wallet I would like to mention that also the (negative) results of rapid antigen tests could (and should) be able to import into Wallet too.

As I think much of the background-processes for creating the wallet passes would be similar, I don't want to create an additional feature-request, with this fairly similar topic, but maybe the rapid test results could be tracked in this thread too.

**Comment by kevin-kraus** [comment]
> As title of this feature-request says that vaccination certificates should be possible to import into Apple Wallet I would like to mention that also the (negative) results of rapid antigen tests could (and should) be able to import into Wallet too.
> 
> As I think much of the background-processes for creating the wallet passes would be similar, I don't want to create an additional feature-request, with this fairly similar topic, but maybe the rapid test results could be tracked in this thread too.

I like your proposal as it would further more improve access speed to the information. Also, Apple Wallet Cards have a "expiring" timestamp AFAIK [See "expirationDate"](https://developer.apple.com/documentation/walletpasses/pass)so they could also be deleted/invalidated automatically. 

I would consider creating a new issue. Your proposal is requesting changes in an other module. Therefore it has no direct connection to this issue.

**Comment by achisto** [comment]
Wouldn't any app that lets your create Apple Wallet Passes be sufficient for solving this problem? There are multiple apps that can create wallet passes from any qr-code (or barcode) you give them. And in the end the qr-code you receive after your vaccination is just that - a qr-code. 🤔

**Comment by Ein-Tim** [comment]
For your information: https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69#issuecomment-860031719 says:

> Unfortunately, this is a security and privacy concern, as already discussed in the CWA GitHub. As this would allow the QR code to be visible without any sort of user validation / passcode. We have discussed this internally and came to the conclusion to not implement it because of the security concerns and that it is out of scope.

So I would not expect any development on this, sorry.

**Comment by kevin-kraus** [comment]
> For your information: https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69#issuecomment-860031719 says:
> 
> > Unfortunately, this is a security and privacy concern, as already discussed in the CWA GitHub. As this would allow the QR code to be visible without any sort of user validation / passcode. We have discussed this internally and came to the conclusion to not implement it because of the security concerns and that it is out of scope.
> 
> So I would not expect any development on this, sorry.

@Ein-Tim you probably could add a disclaimer that privacy is not ensured by using this feature. 

Otherwise what I'm asking myself now, the privacy is also not ensured when a user is using a device not secured by any passcode / biometric authentication. Or am I wrong here?
Therefore a user with no passcode should not be allowed to use the vaccination / rapid test features, right?

**Comment by Ein-Tim** [comment]
@kevin-kraus 

Since I'm also only a community member, I suggest that you add your comment to https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69.

> Otherwise what I'm asking myself now, the privacy is also not ensured when a user is using a device not secured by any passcode / biometric authentication. Or am I wrong here?
Therefore a user with no passcode should not be allowed to use the vaccination / rapid test features, right?

I think the difference here is that the user can protect this data if he wants to, but he can't protect the Apple Wallet even if he wants to...

**Comment by jucktnich** [comment]
@muuuh 1. You can disable access from Lock Screen and 2. you can only access cards which work with Apple Pay from the lockscreen https://support.apple.com/en-us/HT204003

**Comment by kevin-kraus** [comment]
Okay from the comments that I'm seeing in [eu-digital-green-certificates/dgca-wallet-app-ios #69](https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69) there will be no implementation of this feature in the EU context. Otherwise will it still be implemented in the CWA? Because I cannot see any direct links between the whole EU-Certificate stuff and the CWA. Even if the EU don't want to implement it, doesn't necessarily mean that it won't be integrated into CWA. 

I think that it will make a good impact for faster usage of the vaccination certificate. Even today we can see that the certificate will be needed to provide proof in more and more places daily (restaurants, events etc.). The CWA only displays the vaccination QR code. Therefore it should be relatively easy to just copy that QR-Code and insert it into a wallet pass. 

So, can we expect an implementation in CWA? @maugst

**Comment by muuuh** [comment]
@jucktnich

I wrote:
> Please note that having to unlock the phone for opening the app and potentially handing out the phone is a privacy issue (reading mails, photos).
> 
> On the other hand, when the vaccination certificate is in the ios wallet, the vaccination information (owners name?) can be accessed by anybody with access to the phone (without need to unlock).

You wrote:
> @muuuh 1. You can disable access from Lock Screen and 2. you can only access cards which work with Apple Pay from the lockscreen https://support.apple.com/en-us/HT204003

Don't get me wrong. I'm in favor of an option to show the vaccination status in Apple Wallet.

I wanted to point out that both in terms of implementation and non-implementation you can argue with "data protection"/security (i.e. confidentiality).

I think that a conscious setting is necessary in which the user is again made aware of the consequences.

Still, I consider an implementation through third-party apps or saving the QR code as lockscreen background as a workaround.

**Comment by jucktnich** [comment]
@muuuh Dunno why i wrote this 😅

**Comment by ezadoo** [comment]
@kevin-kraus 

> I would consider creating a new issue. Your proposal is requesting changes in an other module. Therefore it has no direct connection to this issue.

As there are currently discussed mainly the privacy concerns and possibilities of integrating the support of Apple Wallet in general I'am refraining from creating a second issue for rapid tests as it would be very similar topic, until a final decision regarding this topic in general is made.

But maybe we could keep the possibility of integrating rapid tests in a similar way too it the back of our minds.

**Comment by iMonZ** [comment]
It would be great if we could clear this up quickly! It’s already pretty late for this feature and if we wait further maybe we don’t need it anymore.
I just don’t see any progression. But vaccinations and negative tests should be in the apple wallet or if this doesn’t work at least as an Apple Watch app!

**Comment by riconeitzel** [comment]
I chime in here! Would be a HUGH step towards broad anticipation! double-plus-thumbs-up

**Comment by Ein-Tim** [comment]
But honestly, if it's not possible to show the QR-Code from the Lock Screen via Apple Wallet (see https://github.com/eu-digital-green-certificates/dgca-wallet-app-ios/issues/69#issuecomment-860042008) then I don't see a benefit that huge... 

Still it would be a nice to have!

**Comment by Ein-Tim** [comment]
FYI: If you want your QR-Code in your Apple Wallet, take a look at: https://twitter.com/kkrdvc/status/1404418854231674885

**Comment by Nils-witt** [comment]
You can choose if can access the wallet when the device is locked.
Here is a screenshot of the setting for it
![IMG_1939615639D5-1](https://user-images.githubusercontent.com/20863269/121897154-5b57e080-cd22-11eb-8372-a2d6874d1a5e.jpeg)

**Comment by jucktnich** [comment]
@Ein-Tim the benefit would be fe that you can use the pass on the watch

**Comment by jucktnich** [comment]
@Nils-witt afaik that's still not possible with cards not using Apple Pay and you also don't have triggers which would lead the pass to appear on the lockscreen as a message

**Comment by Nils-witt** [comment]
Yes, there is no notification or similar to direct access a specific card in this case. But with double click on the home button the wallet requires biometrics(depending on your settings) before it shows any cards and then you can browse all of your cards.
